tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.cc

```diff
@@ -509,25 +518,35 @@ Status TrtShapeOptimizationProfile::RestoreProfiles(
   int K = n_bindings / n_profiles;
 #endif
   int n_inputs = GetNumberOfEngineInputs(engine);
+  if (n_inputs > n_network_inputs) {
+    return errors::Internal("Incorrect number of engine inputs");
+  }
   VLOG(2) << "Attempting to restore " << n_profiles << " profiles, each with "
           << n_inputs << " inputs";
-  SetShapeTensorMask(engine, n_inputs);
+  SetShapeTensorMask(engine, n_network_inputs);
+  is_pruned_input_.resize(n_network_inputs);
```

Comment:
Let's add a comment to the definition of profiles_, to document that if an input, say i,  is pruned, and the profiles is restored from an build, then profiles_[][i] is not initialized.